### PR TITLE
Enable Custom Cipher Selection for Redfish Modules

### PR DESCRIPTION
**Pull request by dlehrman**
##### SUMMARY
Adds an option for ciphers to the redfish modules and aligns with the ciphers option of the ansible.builtin.uri and community.crypto.get_certificate modules.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- redfish_command
- redfish_config
- redfish_info

##### ADDITIONAL INFORMATION
I use Ansible to manage various device types (e.g. PDU's, BMC's, etc) that only support legacy ciphers and/or do not support >2048 bit certificates. For some of these devices, depending on the Ansible controller, the redfish modules were experiencing a handshake failure, though I could still interact them with CLI or GUI web browsers (curl, Chrome, Firefox).

Without manually setting ciphers (example endpoint uses 2048 bit cert, TLSv1.2 / ECDHE-RSA-AES256-SHA, cannot be changed):
```
TASK [community.general.redfish_info] ************************************************************************************************************************************************************************************************************
fatal: [REDACTED]: FAILED! => {"changed": false, "msg": "URL Error on GET request to 'https://REDACTED/redfish/v1/': '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1007)'"}
```

After manually setting ciphers to "HIGH":
```
TASK [community.general.redfish_info] ************************************************************************************************************************************************************************************************************
ok: [REDACTED]

```

Relevant task:
```yaml
    - community.general.redfish_info:
        category: Sessions
        command: GetSessions
        baseuri: "{{ ipv4_address }}"
        username: "{{ username }}"
        password: "{{ password }}"
        ciphers:
          - "HIGH"
```

Ansible controller properties:
- Ubuntu 22.04.4 LTS
- Ansible 2.17.0
- Python 3.10.6-1~22.04
- OpenSSL 3.0.2-0ubuntu1.15

### CODE REVIEW

**Comment by dlehrman** [comment]
The failing CI tests are complaining about `FATAL: A Python version was not specified for environment`, which seems to be an issue with the CI test configuration.

File: plugins/modules/redfish_command.py [hunk 1]
```diff
@@ -302,6 +302,16 @@
...
+  ciphers:
+    required: false
+    description:
+      - SSL/TLS Ciphers to use for the request.
+      - 'When a list is provided, all ciphers are joined in order with V(:).'
+      - See the L(OpenSSL Cipher List Format,https://www.openssl.org/docs/manmaster/man1/openssl-ciphers.html#CIPHER-LIST-FORMAT)
+        for more details.
+      - The available ciphers is dependent on the Python and OpenSSL/LibreSSL versions.
+    type: list
+    elements: str
```
Comment (felixfontein) [inline plugins/modules/redfish_command.py:314]: Same for the others:
```suggestion
    elements: str
    version_added: 9.2.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by felixfontein** [comment]
> The failing CI tests are complaining about `FATAL: A Python version was not specified for environment`, which seems to be an issue with the CI test configuration.

ansible-core devel updated their supported platforms (https://github.com/ansible/ansible/commit/dad6f077319d993c0fa440dd426174a54f034c53), I'll need to adjust CI to that...

**Comment by mraineri** [comment]
The code changes themselves look good to me though; thanks for submitting this!

**Comment by felixfontein** [comment]
@dlehrman thanks for your contribution!
@mraineri thanks for reviewing!

**Pull request merged**